libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1321,6 +1321,7 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 				valueComponentType->category() == Type::Category::RationalNumber &&
 				dynamic_cast<RationalNumberType const&>(*valueComponentType).isFractional() &&
 				valueComponentType->mobileType()
+				// Add Check here (?)
```

Comment:
This is only used to generate a nice error message. What is actually happening here is that in line 1313, we check for an implicit conversion to the variable's type. This type is explicit and not somehow generated from the combination of other types, so this case is perfectly fine.